### ISSUE TITLE
[A11Y] Main desktop navigation and main mobile navigation are not accessible to screen reader users

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#5278]
**Description of Issue**: Main desktop and mobile navigation is not coded properly for screen reader users.
**URL**: https://pr3750-llnmetezlaix5oe10ddq5m3nya7zdz2b.tugboat.qa/
**Testing Environment/Browser**: Windows Lenovo PC, Windows 10 Pro, Chrome Browser v. 94.0.4606.61, JAWS 2020.12.13
**[Severity](https://github.com/backdrop/backdrop-issues/issues/5244#issuecomment-931739260)**: P0
**Solution**: 

1) remove `aria-expanded`/`aria-haspopup` attributes from anchor tags
1) convert the `span` element with class `sub-arrow` to a `button` element.
1) remove `role="group"`, `aria-expanded` attributes, and the `aria-hidden` attribute from inner `<ul>` elements

The aria-expanded attribute is used incorrectly on the expanded (shown/hidden) sections in the main navigation. This attribute should not be used on the expanded grouping (and it should only be used on the top level button that opens the hidden content). 

This main desktop navigation, and the mobile navigation, should use an accordion component structure and ARIA attributes to be accessible to screen readers when content opens and closes. Resources for accessible accordions: https://www.w3.org/TR/wai-aria-practices-1.2/#accordion 

For both the mobile and desktop navigation, the top level link can remain as is, but a button should be used to let users to expand/open the submenu. Please change the code to something like the following:

Top link area:
```
<nav role="navigation">
  <ul>
     <li>
        <a href="/about">About us<a> // remove aria-expanded/aria-haspopup attributes here
         <button aria-label="About us subnavigation" aria-expanded="true/false">+</button> // must use a button here
    </li>
    .....
   </ul>
</nav>
```

For the expanded content below each section, change the code in the following ways:
```
<ul> // remove role="group", aria-expanded attributes, and the aria-hidden attribute. (CSS display: none/block takes the same role as this)
 ....
</ul>
```

For the mobile navigation, please use a button as the top level element that expands the navigation, not a checkbox. Program this button as an accordion with the aria-expanded attribute and ensure that the new content expanded is directly below the button. (resource for accordions is below).

**Screenshot:**
![image](https://user-images.githubusercontent.com/12848168/135544138-17ecf3a8-fa93-46a1-a935-6922f2e8a42c.png)
![image](https://user-images.githubusercontent.com/12848168/135544331-2e687dfe-ffd6-441d-a3c9-0c9bd7d11ac2.png)

**Comment by bhuva-v** [comment]
I'm working on this one

**Comment by stpaultim** [comment]
@bhuva-v - Welcome to the Backdrop issue queue. Please, feel free to ask questions right here in the issue queue.

**Comment by bhuva-v** [comment]
The voice over and screen reader on Mac with Chrome v 94.0.4606.61 was working fine. I do not have a Windows machine with JAWS and will not be able to test the issue. So picking up another one. Please un-assign this one. Thanks

**Comment by jenlampton** [comment]
I'm having a really hard time finding where these attributes are added in Backdrop. I expect most of them are coming from [smartmenus](https://github.com/vadikom/smartmenus), and the changes may need to be made upstream.  

@bhuva-v I would recommend choosing a different issue to work on. This is going to be a tough one!
